dotnet/src/Functions/Functions.OpenApi/OpenApi/OperationSelectionPredicateContext.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.SemanticKernel.Plugins.OpenApi;
+
+/// <summary>
+/// Represents the context for an operation selection predicate.
+/// </summary>
+public readonly record struct OperationSelectionPredicateContext
```

Comment:
I think we were advised to avoid record because of the overhead involved in using them. @westey-m do you have more information about this?

Reply:
We generally avoid records in the core .NET libraries for a few reasons:
1. They add a lot of additional code to the implementation that generally won't be trimmed due to it implementing common interfaces, overriding ToString, etc.
2. We like having control over the surface area, and we lose control with records, since the compiler adds a bunch of stuff.
3. For record classes, (2) includes public surface area that's unspeakable in C# but that we need to think about from the perspective of other languages, namely `<Clone>$`.
4. Semantics when the types contain collections are squishy.

If you're ok with these, using records is ok.